config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_alloydbinstances.alloydb.cnrm.cloud.google.com.yaml

```diff
@@ -121,10 +121,6 @@ spec:
                   instance, instance is created in a random zone with available capacity.
                 type: string
               instanceType:
-                description: 'Immutable. The type of the instance. If the instance
```

Comment:
This field is still immutable, I'm guessing, we just didn't want the `depends_on` terraform-ism in the docs (I'm guessing)?

(If it's not immutable, I have questions about what happens if you go from READ_POOL -> PRIMARY)

Reply:
Yes, the field is still immutable. So modifying the instanceType deletes and recreates the instance.
However, there is no use case where a user needs to modify the instanceType directly.

The READ_POOL do not change their type, and for PRIMARY and SECONDARY instances it always reflects the cluster type of the cluster they are associated with.

We do have `promotion` implementation on the secondary clusters.
For promotion we modify the cluster_type of secondary cluster from SECONDARY to PRIMARY, which internally also modifies the secondary instance to be a primary instance.
And for the same reason we are using instanceTypeRef implemented in https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/1279/files#diff-ea2ee6386ff7f479f831a2b6e3fad19501c9873cb8a9127c33d94e957f3a5ed7R194-R200

Reply:
(Just some non-blocking thoughts.)

Based on the use cases it supports, the field's mutability is kinda tricky: Users are not allowed to update it, but the API will update it when needed, and the users need to keep up with the updated, new value in their gitops flow. Not having control over the fields they set previously seems to be not very well compatible with the declarative style, because it means ordering of imperative operations (who sets the value first, second, third) matters a lot.

Reply:
Just to check - can the user ever set this?  It sounds like it is determined by the cluster?